tests/test_envrc.py

```diff
@@ -0,0 +1,52 @@
+from itertools import permutations
+import os
+
+# Define the cookiecutter directory
+DIR_COOKIECUTTER = "{{ cookiecutter.repo_name }}"
+
+# Initialise an empty list to store the expected filepaths
+DIRS_EXPECTED = []
+
+# Get all the directories within `DIR_COOKIECUTTER`, including all subfolders (unless they are `docs` subfolders)
+for root, dirs, files in os.walk(DIR_COOKIECUTTER):
+    if not os.path.join(DIR_COOKIECUTTER, "docs") in root:
+        DIRS_EXPECTED += list(map(lambda d: os.path.join(os.getcwd(), root, d), dirs))
```

Comment:
```suggestion
    if (not os.path.join(DIR_COOKIECUTTER, "docs") in root) and (not root.endswith("__pycache__")):
        DIRS_EXPECTED += list(map(lambda d: os.path.join(os.getcwd(), root, d), filter(lambda d: d !="__pycache__", dirs)))
```

To exclude `__pycache__` sub-directories.

Reply:
Will expand on this to ensure any root that contains `__pycache__` is not returned here. Marking as **Resolved**.